### PR TITLE
[SYCL][NFC] Remove workaround for SPIR-V translator issues

### PR DESCRIPTION
**Pull request by bader**

### CODE REVIEW

**Comment by bader** [comment]
@intel/llvm-reviewers-runtime, @rbegam, ping.

**Review by alexbatashev — APPROVED** [review]

File: sycl/test/basic_tests/boolean.cpp [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // TODO: Enable compilation w/o -fno-sycl-early-optimizations option.
 // See https://github.com/intel/llvm/issues/2264 for more details.
+// XFAIL: gpu && (level_zero || opencl)
```
Comment (s-kanaev) [inline sycl/test/basic_tests/boolean.cpp:3]: Is there any plan to fix these `XFAIL`'s ?

**Review by s-kanaev — APPROVED** [review]
The changes look good. Though, I have only a single question to placate my curiosity.

File: sycl/test/basic_tests/boolean.cpp (hunk 1 above)
Comment (bader) [reply sycl/test/basic_tests/boolean.cpp:3]: Yes. It's tracked here: https://github.com/intel/llvm/issues/2264.

**Pull request merged**